### ISSUE TITLE
Action invocation history

### ISSUE THREAD

**Issue by jvmlet** [spoike/refluxjs#321]
Hello
I've noticed that action listener is triggered with action invocations that happend *BEFORE* listener has been registered with action.
In below sample the action *testAction* is triggered 3 times *before*  listener is added and the result of 
`fire` function is :
```javascript
fired from click
On document ready 1
On document ready 2
On document ready 3
```
I would expect to see only `fired from click` gets printed. Is it by-design behavior ?
  
The  sample :
```javascript
   <script>
    define('app',['jquery','reflux'],function($,Reflux){
		var testAction = Reflux.createAction();
		$( document ).ready(function() {
			testAction( "On document ready 1" );
			testAction( "On document ready 2" );
			testAction( "On document ready 3" );
			testAction.listen( function(msg) {
				console.debug(msg)
			});
		});
		return {
			action:testAction
		}
	})
	
	function fire(){
		require(['app'],function(app){
			app.action("fired from click");
		})
	}
</script>
<button onClick="fire()">Fire</button>
```

**Comment by spoike** [comment]
Actions are by default deferred. What is happening in the code sample is that the `listen` is performed before the actions are firing off their events. This is by design.

For the behavior you expected, the `listen` call needs to be deferred, e.g. with a `setTimeout` like this:

```javascript
// wrap in setTimeout
window.setTimeout(
    // from the code in OP
    testAction.listen( function(msg) {
      console.debug(msg);
    });
, 13); // thirteen is a good number I guess
```

**Comment by jvmlet** [comment]
Thanks for the detailed explanation

**Comment by jvmlet** [comment]
As a second thought, I would expect to see first 3 action invocations right after document was loaded, *before* button was clicked, but they are fired *only after* the button has been clicked.
